dice_ml/explainer_interfaces/explainer_base.py

```diff
@@ -88,6 +88,9 @@ def generate_counterfactuals(self, query_instances, total_CFs,
         elif isinstance(query_instances, Iterable):
             query_instances_list = query_instances
         for query_instance in tqdm(query_instances_list):
+            # Remaps continuous feature indices based on the query instance
+            self.data_interface.continuous_feature_indexes = [query_instance.columns.get_loc(name) for name in
+                                                              self.data_interface.continuous_feature_names]
```

Comment:
Why is this logic moved here? We are setting a property in public data interface elsewhere. Can we make this a function in public data interface and private data interface and call from here?

Reply:
Done